modules/P07_firmware_bin_base_analyzer.sh

```diff
@@ -15,48 +15,108 @@
 P07_firmware_bin_base_analyzer() {
 
   module_log_init "${FUNCNAME[0]}"
-  module_title "Binary firmware OS detection"
+  module_title "Binary firmware basic analyzer"
 
   os_identification
+
+  # we only do this if we have not found a LInux filesystem
+  if ! [[ -d "$FIRMWARE_PATH" ]]; then
+    binary_architecture_detection
+  fi
 }
 
 os_identification() {
+  sub_module_title "OS detection"
+
+  print_output "[*] Initial OS detection running " | tr -d "\n"
+  print_output "." | tr -d "\n"
```

Comment:
Suggestion: Write a small function for `print_output "." | tr -d "\n"` in `print.sh`. These dots also don't have to be written into log. Out Log looks quite stupid with these dots :D 
Excerpt:
```
[[0;33m*[0m] Initial OS detection running 
.
.
.
.
.
.
.
.
```

Reply:
Also please replace `print_output "." | tr -d "\n"` with the new function call in the `P09` module. Thanks.

Reply:
woo .. thats really ugly. Let's switch to an "echo" for the dots